src/App.jsx

```diff
@@ -1,27 +1,52 @@
 import './App.css';
-import React from 'react';
-
-import RegisterLoginModal from '/src/components/RegisterLoginModal/RegisterLoginModal';
 
 import { ThemeProvider } from '@mui/material/styles';
+import React, { useCallback, useEffect, useState } from 'react';
 import { BrowserRouter, Route, Routes } from 'react-router-dom';
 
 import SellHousePage from '/src/pages/SellHousePage/SellHousePage';
 import theme from '/src/theme/theme';
 
+import { readAll } from './utils/api';
+
 function App() {
+  const [houses, setHouses] = useState(null);
+  const [error, setError] = useState(false);
+  const [loading, setLoading] = useState(true);
   const [LoggedIn, setLoggedIn] = React.useState(false);
+
+  const handleAsyncAction = React.useCallback(async (asyncAction) => {
```

Comment:
Consider using just `useCallback` to make it consistent with other hook calls in this file, and removing the `React` import.